src/basic/virt.c

```diff
@@ -485,6 +485,50 @@ int detect_virtualization(void) {
         return r;
 }
 
+int detect_private_users(void) {
+        static thread_local int cached = -1;
+
+        if (cached < 0) {
+                _cleanup_fclose_ FILE *f = NULL;
+                _cleanup_free_ char *buf = NULL;
+                size_t n_allocated = 0;
+                ssize_t n;
+                uint32_t a, b, c;
+                int r;
+
+                f = fopen("/proc/self/uid_map", "re");
+                if (!f)
+                        return log_debug_errno(errno, "Failed to open /proc/self/uid_map: %m");
+
+                n = getline(&buf, &n_allocated, f);
```

Comment:
i'd probably have implemented the whole function in a much simpler, but less forgiving way: just read the full file with read_full_file() and compare the result literally with "         0          0 4294967295" and be done with it...